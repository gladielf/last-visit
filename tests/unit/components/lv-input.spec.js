import { shallowMount } from '@vue/test-utils';
import LvInput from '@/components/lv-input.vue';

const directives = {
    validate: jest.fn()
};

describe('LvInput - Snapshot', () => {
    it('Snapshot', () => {
        const wrapper = shallowMount(LvInput, {
            directives
        });
        expect(wrapper.element).toMatchSnapshot();
    });
});

describe('LvInput - methods', () => {
    it('btnClicked', async () => {
        const wrapper = shallowMount(LvInput, {
            directives
        });

        wrapper.vm.emitValue({
            target: {
                value: 'foo'
            }
        });

        await wrapper.vm.$nextTick();

        expect(wrapper.emitted('lv-input-value')).toBeTruthy();
    });
});
