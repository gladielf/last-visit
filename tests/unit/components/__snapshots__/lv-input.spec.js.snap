// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LvInput - Snapshot Snapshot 1`] = `
<div
  class="lv-input"
>
  <input
    class="lv-input__input"
    type="text"
  />
  <label
    class="lv-input__error"
  >
    
  </label>
</div>
`;
